main.py

```diff
@@ -117,22 +117,29 @@ def bindDropdownWidget(widget, var):
 		bindDropdownWidget(self.controller2, 'controller_two')
 		bindDropdownWidget(self.controller3, 'controller_three')
 		bindDropdownWidget(self.controller4, 'controller_four')
-		bindFilePicker(self.setXmu1A, self.xmu1APath)
-		bindTextWidget(self.xmu1APath, 'xmu_1a_path')
-		bindFilePicker(self.setXmu1B, self.xmu1BPath)
-		bindTextWidget(self.xmu1BPath, 'xmu_1b_path')
-		bindFilePicker(self.setXmu2A, self.xmu2APath)
-		bindTextWidget(self.xmu2APath, 'xmu_2a_path')
-		bindFilePicker(self.setXmu2B, self.xmu2BPath)
-		bindTextWidget(self.xmu2BPath, 'xmu_2b_path')
-		bindFilePicker(self.setXmu3A, self.xmu3APath)
-		bindTextWidget(self.xmu3APath, 'xmu_3a_path')
-		bindFilePicker(self.setXmu3B, self.xmu3BPath)
-		bindTextWidget(self.xmu3BPath, 'xmu_3b_path')
-		bindFilePicker(self.setXmu4A, self.xmu4APath)
-		bindTextWidget(self.xmu4APath, 'xmu_4a_path')
-		bindFilePicker(self.setXmu4B, self.xmu4BPath)
-		bindTextWidget(self.xmu4BPath, 'xmu_4b_path')
+
+		controller_arr = [
+			['controller_one', self.controller1],
+			['controller_two', self.controller2],
+			['controller_three', self.controller3],
+			['controller_four', self.controller4]
+		]
+
+		for i, controller in enumerate(controller_arr, 1):
+			bindDropdownWidget(eval('self.controller' + str(i)), controller[0])
+			for ab in ['A', 'B']:
+				bindFilePicker(eval('self.setXmu' + str(i) + ab), eval('self.xmu' + str(i) + ab + 'Path'))
+				bindTextWidget(eval('self.xmu' + str(i) + ab + 'Path'), 'xmu_' + str(i) + ab.lower() + '_path')
+
+				if settings.settings[controller[0]] == 'Not connected':
+					eval('self.setXmu' + str(i) + ab + '.setEnabled(False)')
+					eval('self.xmu' + str(i) + ab + 'Path.setEnabled(False)')
+					eval('self.xmuLabel' + str(i) + ab + '.setEnabled(False)')
+
+				eval("controller[1].currentIndexChanged.connect(lambda: self.setXmu" + str(i) + ab + ".setEnabled(True if settings.settings[controller[0]] != 'Not connected' else False))", {'self': self, 'controller': controller, 'settings': settings})
+				eval("controller[1].currentIndexChanged.connect(lambda: self.xmu" + str(i) + ab + "Path.setEnabled(True if settings.settings[controller[0]] != 'Not connected' else False))", {'self': self, 'controller': controller, 'settings': settings})
+				eval("controller[1].currentIndexChanged.connect(lambda: self.xmuLabel" + str(i) + ab + ".setEnabled(True if settings.settings[controller[0]] != 'Not connected' else False))", {'self': self, 'controller': controller, 'settings': settings})
+
```

Comment:
It's code based on my work and when I first proposed it I was trying to figure out a way to avoid a large if/else block & repeating `set.blah.setEnabled(False)`. I think 06f8f44 is much more readable now.

Reply:
That appears to do something different though? This one uses an event listener (`currentIndexChanged`) whereas the other one seems to depend on a hack to trigger an update.